File: identhub/src/main/java/de/solarisbank/identhub/domain/contract/FetchPdfUseCaseFactory.java
package de.solarisbank.identhub.domain.contract;

import de.solarisbank.identhub.file.FileController;
import de.solarisbank.sdk.core.di.internal.Factory;
import de.solarisbank.sdk.core.di.internal.Preconditions;
import de.solarisbank.sdk.core.di.internal.Provider;

public class FetchPdfUseCaseFactory implements Factory<FetchPdfUseCase> {

    private final Provider<ContractSignRepository> contractSignRepositoryProvider;
    private final Provider<FileController> fileControllerProvider;

    public FetchPdfUseCaseFactory(Provider<ContractSignRepository> contractSignRepositoryProvider,
                                  Provider<FileController> fileControllerProvider) {
        this.contractSignRepositoryProvider = contractSignRepositoryProvider;
        this.fileControllerProvider = fileControllerProvider;
    }

    public static FetchPdfUseCaseFactory create(Provider<ContractSignRepository> contractSignRepositoryProvider,
                                                Provider<FileController> fileControllerProvider) {
        return new FetchPdfUseCaseFactory(contractSignRepositoryProvider, fileControllerProvider);
    }

    @Override
    public FetchPdfUseCase get() {
        return Preconditions.checkNotNull(
                new FetchPdfUseCase(contractSignRepositoryProvider.get(), fileControllerProvider.get()),
                "Cannot return null from provider method"
        );
    }
}

File: identhub/src/main/java/de/solarisbank/identhub/domain/di/contract/FetchPdfUseCaseFactory.kt
package de.solarisbank.identhub.domain.di.contract

import de.solarisbank.identhub.data.contract.ContractSignRepository
import de.solarisbank.identhub.domain.contract.FetchPdfUseCase
import de.solarisbank.identhub.file.FileController
import de.solarisbank.sdk.feature.di.internal.Factory
import de.solarisbank.sdk.feature.di.internal.Preconditions
import de.solarisbank.sdk.feature.di.internal.Provider

class FetchPdfUseCaseFactory(
    private val contractSignRepositoryProvider: Provider<ContractSignRepository>,
    private val fileControllerProvider: Provider<FileController>
) : Factory<FetchPdfUseCase> {
    override fun get(): FetchPdfUseCase {
        return Preconditions.checkNotNull(
            FetchPdfUseCase(contractSignRepositoryProvider.get(), fileControllerProvider.get()),
            "Cannot return null from provider method"
        )
    }

    companion object {
        @JvmStatic
        fun create(
            contractSignRepositoryProvider: Provider<ContractSignRepository>,
            fileControllerProvider: Provider<FileController>
        ): FetchPdfUseCaseFactory {
            return FetchPdfUseCaseFactory(contractSignRepositoryProvider, fileControllerProvider)
        }
    }
}